File: skeleton/src/main/java/com/example/routing/domain/model/EtaState.java
package com.example.routing.domain.model;

import java.time.Instant;

public record EtaState(
    long remainingSeconds,
    Instant calculatedAt,
    double confidence,
    double smoothedSpeedMps,
    Instant lastLocationAt,
    boolean degraded
) {
    public static EtaState initial(Instant now) {
        return new EtaState(0, now, 0.0, 0.0, now, true);
    }
}

File: pilot_app/lib/core/domain/value_objects/eta_state.dart
/// Estado do ETA (tempo restante, confiança, degradado). Doc 02 / 07.
class EtaState {
  const EtaState({
    required this.remainingSeconds,
    required this.calculatedAt,
    required this.confidence,
    required this.smoothedSpeedMps,
    required this.lastLocationAt,
    required this.degraded,
  });

  final int remainingSeconds;
  final DateTime calculatedAt;
  final double confidence;
  final double smoothedSpeedMps;
  final DateTime lastLocationAt;
  final bool degraded;
}
